spec/unit/pdk/validate/puppet/puppet_environment_conf_validator_spec.rb

```diff
@@ -0,0 +1,158 @@
+    it { is_expected.to eq('puppet-environment-conf') }
+  end
+
+  describe '.spinner_text' do
+    subject(:spinner_text) { validator.spinner_text }
+
+    it { is_expected.to match(%r{Checking Puppet Environment settings}i) }
+  end
+
+  describe '.valid_in_context?' do
+    subject(:valid_context) { validator.valid_in_context? }
+
+    context 'in a Control Repo' do
+      let(:validator_context) { PDK::Context::ControlRepo.new(nil, nil) }
+
+      it { is_expected.to eq(true) }
+    end
+
+    context 'in a Module' do
+      let(:validator_context) { PDK::Context::Module.new(nil, nil) }
+
+      it { is_expected.to eq(false) }
+    end
+
+    context 'in an unknown context' do
+      let(:validator_context) { PDK::Context::None.new(nil) }
+
+      it { is_expected.to eq(false) }
+    end
+  end
+
+  describe '.validate_target' do
+    subject(:return_value) { described_class.new.validate_target(report, target[:name]) }
+
+    let(:report) { PDK::Report.new }
+
+    before(:each) do
+      [
+        target[:name],
+        File.join(validator_context.root_path, target[:name]),
+      ].each do |filename|
+        allow(PDK::Util::Filesystem).to receive(:directory?).with(filename).and_return(target.fetch(:directory, false))
+        allow(PDK::Util::Filesystem).to receive(:file?).with(filename).and_return(target.fetch(:file, true))
+        allow(PDK::Util::Filesystem).to receive(:readable?).with(filename).and_return(target.fetch(:readable, true))
+        allow(PDK::Util::Filesystem).to receive(:read_file).with(filename).and_return(target.fetch(:content, ''))
+      end
+    end
+
+    context 'when a target is provided that is an unreadable file' do
+      let(:target) { { name: 'environment.conf', readable: false } }
+
+      it 'adds a failure event to the report' do
+        expect(report).to receive(:add_event).with(
+          file:     target[:name],
+          source:   validator.name,
+          state:    :failure,
+          severity: 'error',
+          message:  'Could not be read.',
+        )
+        expect(return_value).to eq(1)
+      end
+    end
+
+    context 'when a target is provided that is not a file' do
+      let(:target) { { name: 'environment.conf', file: false } }
+
+      it 'skips the target' do
+        expect(report).not_to receive(:add_event)
+      end
+    end
+
+    context 'when a target is provided that is empty' do
+      let(:target) { { name: 'environment.conf', content: '' } }
+
+      it 'adds a passing event to the report' do
+        expect(report).to receive(:add_event).with(
+          file:     target[:name],
+          source:   validator.name,
+          state:    :passed,
+          severity: 'ok',
+        )
+        expect(return_value).to eq(0)
+      end
+    end
+
+    context 'when a target is provided that has all allowed settings and is valid' do
+      let(:target) { { name: 'environment.conf', content: "modulepath=foo\nmanifest=foo\nconfig_version=foo\nenvironment_timeout=0" } }
```

Comment:
Maybe move `content` to it's own var/`let` block and use a heredoc to make it easier to maintain?

Reply:
I managed to use a heredoc inside the hash which looks ok e.g.

``` ruby
      let(:target) do
        {
          name: 'environment.conf',
          content: <<-EOT
                   modulepath=foo
                   manifest=foo
                   config_version=foo
                   environment_timeout=0
                   EOT
        }
      end
```